turbopack/crates/turbo-tasks-auto-hash-map/src/map.rs

```diff
@@ -900,6 +901,25 @@ where
     }
 }
 
+impl<K, V, H, const I: usize> ShrinkToFit for AutoMap<K, V, H, I>
+where
+    K: Eq + Hash,
+    V: Eq,
+    H: BuildHasher + Default,
+{
+    fn shrink_to_fit(&mut self) {
+        match self {
+            AutoMap::List(list) => list.shrink_to_fit(),
+            AutoMap::Map(map) => {
+                if map.len() < MIN_HASH_SIZE {
+                    self.convert_to_list();
```

Comment:
Looks like this creates a `SmallVec::with_capacity(MAX_LIST_SIZE)`, which might be too large?

So rather convert first, if necessary, and then try to shrink whatever format it is in?